plugins/test/unit/plugins/importers/yum/test_purge.py

```diff
@@ -288,3 +290,23 @@ def test_retain_old(self, mock_remove_old_versions):
         purge.purge_unwanted_units(self.metadata_files, self.conduit, self.config)
 
         mock_remove_old_versions.assert_called_once_with(3, self.conduit)
+
+
+class RemoveUnitDuplicateNevra(TestPurgeBase):
+
+    @mock.patch.object(purge, 'RepoUnitAssociationManager', autospec=True)
+    @mock.patch.object(purge, 'UnitAssociationCriteria', autospec=True)
+    def test_remove_unit_duplicate_nerva(self, mock_criteria, mock_association):
+        unit_key = {'name': 'test-nevra', 'epoch': 0, 'version': 1, 'release': '23',
+                    'arch': 'noarch', 'checksum': '1234abc', 'checksumtype': 'sha256'}
+        type_id = 'rpm'
+        repo_id = 'test-repo'
+        expected_filters ={'$and': [{'epoch': 0}, {'version': 1}, {'name': 'test-nevra'},
+                                    {'release': '23'}, {'arch': 'noarch'}]}
+
+        purge.remove_unit_duplicate_nevra(unit_key, type_id, repo_id)
+
+        mock_criteria.assert_called_once_with(unit_filters=expected_filters, type_ids=type_id)
```

Comment:
The order of these filters is not guaranteed since they are generated from a dictionary's items() method. This means that this test could fail on alternative (or future) versions or implementations of Python.

I'd recommend using a set on the mock_criteria.mock_calls[0] to determine which filters were applied, and then comparing that with an expected set to ensure they are equal. This way, the test will pass no matter which order the dictionary's items are returned.